### DISCUSSION TITLE
Software Reset / Reboot the running program on bluepill (STM32F103C8T6)

### DISCUSSION THREAD

**Discussion by HappySamuel** [stm32duino/Arduino_Core_STM32#1476] [Q&A] [answered]
Hi

As we know, pressing reset button on the bluepill board, the running software will be rebooted.

I would like to do the same in software method, resetting of running program on the bluepill board (STM32F103C8T6). Description: 
I send a reset msg to the board via serial communication, variable name "reboot" is triggered to true, and when the if function detect "reboot" as true, it will reboot the board.

How shall i achieve this? What command /library is needed to achieve this?

Note: I am using Arduino IDE

Best,
Samuel

**Comment by fpistm** [comment, marked as answer]
Hi @HappySamuel 
You can use:
`NVIC_SystemReset();`


http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dai0321a/BIHDHCGJ.html

**Reply by SeongJongKwak** [reply]
@fpistm  Dear fpistm

Error msg : 
'NVIC_SystemReset' was not declared in this scope

STM32 Bluepill on arduino IDE.
If so, can you tell me why or how?
Or, is there a library?

**Reply by fpistm** [reply]
You probably not use this core but roger's one.

**Reply by SeongJongKwak** [reply]
NVIC_SystemReset();
or
nvic_sys_reset();
or
HAL_NVIC_SystemReset();

**Comment by psystein** [comment]
Hey @fpistm, 

I have been using arduino IDE for AtMega and ESP. 
Now venturing into STM32 Bluepill on arduino IDE. 

Tried using the  `NVIC_SystemReset();`  but the code doesn't compile. 
Vebrose Output Says : 
`exit status 1
'NVIC_SystemReset' was not declared in this scope
`

What am i doing wrong?

**Reply by fpistm** [reply]
Are you sure you use this core ?

**Reply by psystein** [reply]
Hello,

I have added thee below link to Board Manager URL in arduino ide preferences. 
http://dan.drown.org/stm32duino/package_STM32duino_index.json

And my board selector shows up like this.. 
![image](https://user-images.githubusercontent.com/45229407/192987172-59607252-4cb9-4479-8fbc-06c705a0dea1.png)

**Reply by psystein** [reply]
Oh.... 
I See my mistake.. 
Am getting onto fixing it now.. 
My mis-understanding..

**Reply by SeongJongKwak** [reply]
@psystein 

Error msg : 
'NVIC_SystemReset' was not declared in this scope

STM32 Bluepill on arduino IDE.
If so, can you tell me why or how?
Or, is there a library?

**Reply by psystein** [reply]
I was using a different core for stm32. 
The function is not defined in that core. Use the correct core and it works.

**Comment by asdrubalfuentes** [comment]
i try with nvic_sys_reset(); and go fine.
